src/compiler/scala/tools/nsc/typechecker/Macros.scala

```diff
@@ -906,33 +910,33 @@ trait Macros extends MacroRuntimes with Traces with Helpers {
   var hasPendingMacroExpansions = false // JZ this is never reset to false. What is its purpose? Should it not be stored in Context?
   def typerShouldExpandDeferredMacros: Boolean = hasPendingMacroExpansions && !delayed.isEmpty
   private val forced = perRunCaches.newWeakSet[Tree]
-  private val delayed = perRunCaches.newWeakMap[Tree, scala.collection.mutable.Set[Int]]()
```

Comment:
huh, what was the point of the `Set[Int]` anyways?

Reply:
I could only guess it was trying to avoid a leak of actual `Symbol`s? As in, a defensive measure against us failing to clear entries from here? I think that's a valid thing to (hypothetically) worry about, at least not now, as I already put a backstop to clear `delayed` completely in `TyperPhase.run`.